components/forms/src/api.rs

```diff
@@ -0,0 +1,553 @@
+            )
+        ",
+        common_cols = COMMON_COLS
+    );
+
+    let address = tx.query_row(&sql, &[guid.as_str()], |row| {
+        let address_fields = NewAddressFields {
+            given_name: row.get("given_name")?,
+            additional_name: row.get("additional_name")?,
+            family_name: row.get("family_name")?,
+            organization: row.get("organization")?,
+            street_address: row.get("street_address")?,
+            address_level3: row.get("address_level3")?,
+            address_level2: row.get("address_level2")?,
+            address_level1: row.get("address_level1")?,
+            postal_code: row.get("postal_code")?,
+            country: row.get("country")?,
+            tel: row.get("tel")?,
+            email: row.get("email")?,
+        };
+
+        Ok(Address {
+            guid: Guid::from_string(row.get("guid")?),
+            address_fields,
+            time_created: row.get("time_created")?,
+            time_last_used: row.get("time_last_used")?,
+            time_last_modified: row.get("time_last_modified")?,
+            times_used: row.get("times_used")?,
+            sync_change_counter: row.get("sync_change_counter")?,
+        })
+    })?;
+
+    tx.commit()?;
+    Ok(address)
+}
+
+#[allow(dead_code)]
+pub fn get_all_addresses(conn: &mut Connection) -> Result<Vec<Address>> {
+    let tx = conn.transaction()?;
+    let mut addresses = Vec::new();
+    let sql = "
+        SELECT
+            guid,
+            given_name,
+            additional_name,
+            family_name,
+            organization,
+            street_address,
+            address_level3,
+            address_level2,
+            address_level1,
+            postal_code,
+            country,
+            tel,
+            email,
+            time_created,
+            time_last_used,
+            time_last_modified,
+            times_used,
+            sync_change_counter
+        FROM addresses_data d
+        WHERE NOT EXISTS (
+            SELECT 1
+            FROM   addresses_tombstones t
+            WHERE  d.guid = t.guid
+        )
+    ";
+
+    {
+        let mut stmt = tx.prepare(sql)?;
+        let addresses_iter = stmt.query_map(NO_PARAMS, |row| {
+            let new_address = NewAddressFields {
+                given_name: row.get("given_name")?,
+                additional_name: row.get("additional_name")?,
+                family_name: row.get("family_name")?,
+                organization: row.get("organization")?,
+                street_address: row.get("street_address")?,
+                address_level3: row.get("address_level3")?,
+                address_level2: row.get("address_level2")?,
+                address_level1: row.get("address_level1")?,
+                postal_code: row.get("postal_code")?,
+                country: row.get("country")?,
+                tel: row.get("tel")?,
+                email: row.get("email")?,
+            };
+            Ok(Address {
```

Comment:
Would you mind factoring this out into an `Address::from_row` helper function that `get_all_addresses` and `get_address` could use?